src/twisted/application/runner/_runner.py

```diff
@@ -31,19 +29,73 @@ class Runner(object):
     log = Logger()
 
 
-    def __init__(self, options):
+    def __init__(
+        self,
+        reactor=None,
+        pidFile=nonePIDFile, kill=False,
+        defaultLogLevel=LogLevel.info,
+        logFile=stderr, fileLogObserverFactory=textFileLogObserver,
+        whenRunning=lambda **_: None, whenRunningArguments={},
+        reactorExited=lambda **_: None, reactorExitedArguments={},
+    ):
         """
-        @param options: Configuration options for this runner.
-        @type options: mapping of L{RunnerOptions} to values
+        @param reactor: The reactor to start and run the application in.
+        @type reactor: L{IReactorCore}
```

Comment:
As implemented here, this `@type` is wrong; sometimes it's `None`, if it's given the default.

Personally I'd rather just eliminate the default and force the caller to pass a usable value.